diff --git a/libavcodec/h264_slice.c b/libavcodec/h264_slice.c
index 1c9a270fb6..a68ada8f8c 100644
--- a/libavcodec/h264_slice.c
+++ b/libavcodec/h264_slice.c
@@ -1463,6 +1463,12 @@ static int h264_field_start(H264Context *h, const H264SliceContext *sl,
         h->avctx->has_b_frames = sps->num_reorder_frames;
     }
 
+    if (sps && sps->bitstream_restriction_flag &&
+        h->avctx->has_b_frames < sps->num_reorder_frames) {
+        h->avctx->has_b_frames = FFMAX(h->avctx->has_b_frames,
+                                       sps->num_reorder_frames);
+    }
+
     last_pic_droppable   = h->droppable;
     last_pic_structure   = h->picture_structure;
     h->droppable         = (nal->ref_idc == 0);
