pipe2() => __pipe2()

diff --git a/tests/check/pipelines/ipcpipeline.c b/tests/check/pipelines/ipcpipeline.c
index b44cdd6fa..436c0c011 100644
--- a/tests/check/pipelines/ipcpipeline.c
+++ b/tests/check/pipelines/ipcpipeline.c
@@ -39,7 +39,7 @@
 
 #ifndef HAVE_PIPE2
 static int
-pipe2 (int pipedes[2], int flags)
+__pipe2 (int pipedes[2], int flags)
 {
   int ret = pipe (pipedes);
   if (ret < 0)
@@ -791,10 +791,10 @@ test_base (const char *name, TestFeatures features,
 
   weak_refs = NULL;
 
-  FAIL_IF (pipe2 (pipesfa, O_NONBLOCK) < 0);
-  FAIL_IF (pipe2 (pipesba, O_NONBLOCK) < 0);
-  FAIL_IF (pipe2 (pipesfv, O_NONBLOCK) < 0);
-  FAIL_IF (pipe2 (pipesbv, O_NONBLOCK) < 0);
+  FAIL_IF (__pipe2 (pipesfa, O_NONBLOCK) < 0);
+  FAIL_IF (__pipe2 (pipesba, O_NONBLOCK) < 0);
+  FAIL_IF (__pipe2 (pipesfv, O_NONBLOCK) < 0);
+  FAIL_IF (__pipe2 (pipesbv, O_NONBLOCK) < 0);
   FAIL_IF (socketpair (PF_UNIX, SOCK_STREAM, 0, ctlsock) < 0);
 
   FAIL_IF (pipesfa[0] < 0);
@@ -814,7 +814,7 @@ test_base (const char *name, TestFeatures features,
   if (features & TEST_FEATURE_RECOVERY_MASTER_PROCESS) {
     /* the other master will let us know its child's PID so we can unwind
        it when we're finished */
-    FAIL_IF (pipe2 (master_recovery_pid_comm, O_NONBLOCK) < 0);
+    FAIL_IF (__pipe2 (master_recovery_pid_comm, O_NONBLOCK) < 0);
 
     recovery_pid = fork ();
     if (recovery_pid > 0) {
@@ -2073,7 +2073,7 @@ GST_START_TEST (test_empty_upstream_query)
   upstream_query_slave_data sd = { {0}
   };
 
-  FAIL_UNLESS (pipe2 (id.ts_pipes, O_NONBLOCK) == 0);
+  FAIL_UNLESS (__pipe2 (id.ts_pipes, O_NONBLOCK) == 0);
   TEST_BASE (TEST_FEATURE_TEST_SOURCE, upstream_query_source,
       setup_sink_upstream_query, check_success_source_upstream_query,
       check_success_sink_upstream_query, &id, &md, &sd);
@@ -2090,7 +2090,7 @@ GST_START_TEST (test_wavparse_upstream_query)
   upstream_query_slave_data sd = { {0}
   };
 
-  FAIL_UNLESS (pipe2 (id.ts_pipes, O_NONBLOCK) == 0);
+  FAIL_UNLESS (__pipe2 (id.ts_pipes, O_NONBLOCK) == 0);
   TEST_BASE (TEST_FEATURE_WAV_SOURCE, upstream_query_source,
       setup_sink_upstream_query, check_success_source_upstream_query,
       check_success_sink_upstream_query, &id, &md, &sd);
@@ -2107,7 +2107,7 @@ GST_START_TEST (test_mpegts_upstream_query)
   upstream_query_slave_data sd = { {0}
   };
 
-  FAIL_UNLESS (pipe2 (id.ts_pipes, O_NONBLOCK) == 0);
+  FAIL_UNLESS (__pipe2 (id.ts_pipes, O_NONBLOCK) == 0);
   TEST_BASE (TEST_FEATURE_MPEGTS_SOURCE, upstream_query_source,
       setup_sink_upstream_query, check_success_source_upstream_query,
       check_success_sink_upstream_query, &id, &md, &sd);
@@ -2124,7 +2124,7 @@ GST_START_TEST (test_mpegts_2_upstream_query)
   upstream_query_slave_data sd = { {0}
   };
 
-  FAIL_UNLESS (pipe2 (id.ts_pipes, O_NONBLOCK) == 0);
+  FAIL_UNLESS (__pipe2 (id.ts_pipes, O_NONBLOCK) == 0);
   TEST_BASE (TEST_FEATURE_MPEGTS_SOURCE | TEST_FEATURE_SPLIT_SINKS,
       upstream_query_source, setup_sink_upstream_query,
       check_success_source_upstream_query, check_success_sink_upstream_query,
@@ -2142,7 +2142,7 @@ GST_START_TEST (test_live_a_upstream_query)
   upstream_query_slave_data sd = { {0}
   };
 
-  FAIL_UNLESS (pipe2 (id.ts_pipes, O_NONBLOCK) == 0);
+  FAIL_UNLESS (__pipe2 (id.ts_pipes, O_NONBLOCK) == 0);
   TEST_BASE (TEST_FEATURE_LIVE_A_SOURCE,
       upstream_query_source, setup_sink_upstream_query,
       check_success_source_upstream_query, check_success_sink_upstream_query,
@@ -2160,7 +2160,7 @@ GST_START_TEST (test_live_av_upstream_query)
   upstream_query_slave_data sd = { {0}
   };
 
-  FAIL_UNLESS (pipe2 (id.ts_pipes, O_NONBLOCK) == 0);
+  FAIL_UNLESS (__pipe2 (id.ts_pipes, O_NONBLOCK) == 0);
   TEST_BASE (TEST_FEATURE_LIVE_AV_SOURCE,
       upstream_query_source, setup_sink_upstream_query,
       check_success_source_upstream_query, check_success_sink_upstream_query,
@@ -2178,7 +2178,7 @@ GST_START_TEST (test_live_av_2_upstream_query)
   upstream_query_slave_data sd = { {0}
   };
 
-  FAIL_UNLESS (pipe2 (id.ts_pipes, O_NONBLOCK) == 0);
+  FAIL_UNLESS (__pipe2 (id.ts_pipes, O_NONBLOCK) == 0);
   TEST_BASE (TEST_FEATURE_LIVE_AV_SOURCE | TEST_FEATURE_SPLIT_SINKS,
       upstream_query_source, setup_sink_upstream_query,
       check_success_source_upstream_query, check_success_sink_upstream_query,
